### ISSUE TITLE
Setting for notes to ignore

### ISSUE THREAD

**Issue by zachdaniel** [hadynz/obsidian-sidekick#3]
Hello!

This plugin is awesome! One thing that could be really useful would be the ability to exclude certain notes or folders from suggestions via settings. For example, I have a note for myself in my CRM that has an alias for `Me` and `Myself` (which is kinda silly, just did it for tinkering), but now any word that contains `me` has that part highlighted 😆  For now I'll just remove that alias, but depending on the subject matter (like chemistry) you might end up with every letter in the alphabet lit up 😆

**Comment by hadynz** [comment]
That's a really good call. Can see how that would be quite beneficial. 

Would adding the following exclusion types achieve what you are trying to do:

* Exclude indexing selected list of folders
* Exclude indexing list of tags

**Comment by zachdaniel** [comment]
Those are both useful, and I can see myself using both. However, the current one that matters to me is excluding a list of notes specifically, (e.g in the case that some note often suggests being linked). In my example, I'd like to be able to tell it to ignore the file called `Me`.

For specific files, it could also be done as yaml frontmatter, e.g

```yaml
tags: ...
aliases: ....
sidekick: false
```

**Comment by chrisgrieser** [comment]
I agree, I also love the idea of the plugin, but a lot of the highlights are just unnecessary. ignoring would make the plugin much more valuable.

**Comment by hadynz** [comment]
> I agree, I also love the idea of the plugin, but a lot of the highlights are just unnecessary. ignoring would make the plugin much more valuable.

Are you able to share some examples of what some of those unnecessary highlights look like?

This will help me as I am working on this feature.

**Comment by chrisgrieser** [comment]
highlighting everything that has a "pr", because I have a tag `#pr` (for public relations), for example.

Also, it have notes on some generic things like [[Dataview]] and it then highlights the word every time I write it, even though it doesn't need to have a link every time.

**Comment by hadynz** [comment]
The issue of partial highlighting (that you get with `#pr`) will be resolved as part of https://github.com/hadynz/obsidian-sidekick/issues/4.

> Also, it have notes on some generic things like [[Dataview]] and it then highlights the word every time I write it, even though it doesn't need to have a link every time.

What would you ideally expect from the plugin when dealing with very generic keywords in your notes and tags (like `Dataview`)?

**Comment by chrisgrieser** [comment]
I think simply having options to ignore them would suffice. Like `sidekick: ignore` in the yaml of `[[Dataview]]`, or a plugin setting to save ignore a folder or sth.

**Comment by HEmile** [comment]
I created a setting in the pull request to allow disabling certain keywords (regardless of where it comes from). You can install it from here until it's pulled upstream. https://github.com/HEmile/obsidian-sidekick/releases/tag/1.1.0
